lib/src/dart_download.dart

```diff
@@ -0,0 +1,250 @@
+      saved = alreadyDownloaded;
+    }
+    IOSink? outStream;
+    var hasError = false;
+    try {
+      final response = await request.close();
+      //print('Response headers:');
+      if (response.statusCode == 200) {
+        response.headers.forEach((name, values) async {
+          //print('- $name: $values');
+          if (name == 'etag') {
+            etag = values.first;
+            //print('has etag');
+            await updateMetaData();
+          } else if (name == 'accept-ranges') {
+            //print('can be continued!');
+            resumable = true;
+          } else if (name == 'content-length') {
+            //print('${values.first} to download');
+            finalSize = int.parse(values.first);
+          }
+        });
+      }
+      // Cancel download after timeout for testing:
+      //Future<void>.delayed(const Duration(milliseconds: 500), () {
+      //  _httpClient.close(force: true);
+      //  _httpClient = _createHttpClient();
+      //  //print('### close called!');
+      //});
+      final mode =
+          response.statusCode == 206 ? FileMode.append : FileMode.write;
+      outStream =
+          cacheFile.openWrite(mode: mode, encoding: Encoding.getByName('l1')!);
+      final counter = StreamTransformer<List<int>, List<int>>.fromHandlers(
+        handleData: (data, sink) {
+          sink.add(data);
+          saved += data.length;
+          if (finalSize != null) {
+            progress = (saved * 1000) ~/ finalSize!;
+          }
+        },
+        handleDone: (sink) async {
+          print('Saved $saved bytes to cache file ${cacheFile.absolute}');
+          if (!hasError) {
+            _status = DownloadStatus.completed;
+          }
+          notifyListeners();
+          await outStream?.flush();
+          await outStream?.close();
+          outStream = null;
+        },
+        handleError: (e, trace, sink) async {
+          print('Error: $e');
+          sink.close();
+          await outStream?.flush();
+          await outStream?.close();
+          outStream = null;
+          hasError = true;
+          _status =
+              resumable == true ? DownloadStatus.paused : DownloadStatus.failed;
+          notifyListeners();
+        },
+      );
+      await response.transform(counter).pipe(outStream!);
+    } on HttpException catch (e, trace) {
+      //print('### error: $e');
+      _status =
+          resumable == true ? DownloadStatus.paused : DownloadStatus.failed;
+      notifyListeners();
+    }
+  }
+
+  /// Pauses the download when running.
+  @override
+  Future<void> pause() async {
+    print('paused called on DartDownload');
+    _httpClient?.close(force: true);
+  }
+
+  /// Cancel the download when running or paused
+  @override
+  Future<void> cancel() async {
+    await pause();
+    await delete();
+    _status = DownloadStatus.canceled;
+    notifyListeners();
+  }
+
+  /// Delete the download
```

Comment:
```suggestion
  /// Deletes this download.
```

What if this download is running? I see in code it's paused first, let's also mention it in the docs.

While Swift/Kotlin documentation is useful mainly for this plugin's maintainers, Dart documentation is quite important because people see it in their IDEs. That's why I'm stressing this so much, sorry! :D